packages/server/src/api/controllers/automation.js

```diff
@@ -268,15 +176,37 @@ module.exports.getDefinitionList = async function (ctx) {
  *********************/
 
 exports.trigger = async function (ctx) {
-  const db = new CouchDB(ctx.appId)
+  const appId = ctx.appId
+  const db = new CouchDB(appId)
   let automation = await db.get(ctx.params.id)
   await triggers.externalTrigger(automation, {
     ...ctx.request.body,
-    appId: ctx.appId,
+    appId,
   })
-  ctx.status = 200
   ctx.body = {
     message: `Automation ${automation._id} has been triggered.`,
     automation,
   }
 }
+
+exports.test = async function (ctx) {
+  const appId = ctx.appId
+  const db = new CouchDB(appId)
+  let automation = await db.get(ctx.params.id)
+  await setTestFlag(automation._id)
+  const response = await triggers.externalTrigger(
+    automation,
+    {
+      ...ctx.request.body,
+      appId,
+    },
+    { getResponses: true }
+  )
+  // save a test history run
+  await updateTestHistory(ctx.appId, automation, {
+    ...ctx.request.body,
+    occurredAt: new Date().toISOString(),
```

Comment:
NAB: Might be better to store this as a timestamp in the DB

Reply:
I thought about using a timestamp here, the only thing is that our system uses ISO strings for everything right now, do we want to make a decision on that?

Personally I think using ISO strings for users dates is good as it includes they selected timezone etc, but I think for storing dates/times that things happened internally we should just use timestamps as they are easier to work with in the DB etc.